ksql-api/src/main/java/io/confluent/ksql/api/server/QueryStreamHandler.java

```diff
@@ -80,10 +79,11 @@ public void handle(final RoutingContext routingContext) {
     createQueryPublisherAsync(queryStreamArgs.get().sql, queryStreamArgs.get().properties, context)
         .thenAccept(queryPublisher -> {
 
-          final PushQueryHolder query = connectionQueryManager
-              .createApiQuery(queryPublisher, routingContext.request());
+          final PushQueryHolder query = queryPublisher.isPullQuery() ? null :
+              connectionQueryManager.createApiQuery(queryPublisher, routingContext.request());
 
-          final QueryResponseMetadata metadata = new QueryResponseMetadata(query.getId().toString(),
+          final QueryResponseMetadata metadata = new QueryResponseMetadata(
+              query == null ? null : query.getId().toString(),
```

Comment:
Can we _please_ not have methods taking and returning `null`. If it's optional, make it `Optional`!  Jackson knows how to deal with `Optional`.

Or just use an empty string for pull query ids and mark `QueryResponseMetadata ` with `@JsonInclude(NON_EMPTY)`, which will mean it doesn't include a `queryId` field in the response if the id is empty.

Reply:
Well, according to Brian Goetz, Optional was _not_ designed to be used in method args (and IntelliJ will, by default, warn you if you do this) ;)

So this is very much a matter of personal style, there is no objective "right" way to do this.

We could argue about this until the cows come home, but imho it's of little importance and we have more important things to spend our time on :)

Reply:
What is your objection to using an empty string?

By using `null` anyone using this type must null check the Id _everytime_ be for accessing it and do something if its null.   This has a tendency to produce NPEs and hence bugs.

Why not just an empty string?

Reply:
Empty string seems hacky to me. The query has no idea, the correct way to represent that inside an object is as a null field (Optionals weren't designed for this case).

Reply:
I have fixed this in a simple way which avoids nulls but doesn't require any new types. I hope you agree this is preferable.